airsonic-main/src/main/java/org/airsonic/player/service/search/AnalyzerFactory.java

```diff
@@ -0,0 +1,140 @@
+/*
+ This file is part of Airsonic.
+
+ Airsonic is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Airsonic is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Airsonic.  If not, see <http://www.gnu.org/licenses/>.
+
+ Copyright 2016 (C) Airsonic Authors
+ Based upon Subsonic, Copyright 2009 (C) Sindre Mehus
+ */
+
+package org.airsonic.player.service.search;
+
+import org.apache.lucene.analysis.ASCIIFoldingFilter;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.LowerCaseFilter;
+import org.apache.lucene.analysis.StopFilter;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.analysis.standard.StandardFilter;
+import org.apache.lucene.analysis.standard.StandardTokenizer;
+import org.apache.lucene.util.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.io.Reader;
+
+/**
+ * Analyzer provider.
+ * This class is a division of what was once part of SearchService and added functionality.
+ * This class provides Analyzer which is used at index generation
+ * and QueryAnalyzer which analyzes the specified query at search time.
+ * Analyzer can be closed but is a reuse premise.
+ * It is held in this class.
+ */
+@Component
+public final class AnalyzerFactory {
+  
+  private static final Logger LOG = LoggerFactory.getLogger(AnalyzerFactory.class);
+
+  private Analyzer analyzer;
+
+  private Analyzer queryAnalyzer;
+
+  /**
+   * Return analyzer.
+   * 
+   * @return analyzer for index
+   */
+  public Analyzer getAnalyzer() {
+    if (null == analyzer) {
+      try {
+        analyzer = new CustomAnalyzer();
+      } catch (Exception e) {
+        /*
+         * It is a transient description, and it is appropriate to detect an IOException at Lucene

```

Comment:
I think you meant "temporary" :)

Reply:
That's right.
Here, a new version of Analyzer is generated after the update.